### PR TITLE
Handshake middleware for server

### PR DESCRIPTION
**Pull request by t3ran13**
Possibility to add own handshake middleware for ws server
For example, I made handshake with OAuth access token)

### CODE REVIEW

**Comment by ProPheT777** [comment]
Hey, 

Thank your for your contribution, could you open a PR on https://github.com/GeniusesOfSymfony/WebsocketAppDemo and target your branch via composer 

```
gos/web-socket-bundle: '1.8.6 as dev-staging-handcheck-middleware'
```

And add a sample handshake middleware then I can run integration against it.

**Comment by t3ran13** [comment]
I do not think that I understand that I have to do (it is my first PR), but I made fix for the constructor and added middleware example to docs (HandshakeMiddleware.md)

**Comment by ProPheT777** [comment]
Staging branch `staging-handcheck-middleware` updated :+1:

**Comment by t3ran13** [comment]
https://github.com/GeniusesOfSymfony/WebsocketAppDemo/pull/7
samle

**Comment by ProPheT777** [comment]
Got this after composer install : 

image

**Comment by t3ran13** [comment]
Sorry, l will look to it in a few days

File: Resources/docs/HandshakeMiddleware.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                new ClientRejectedEvent($e->getMessage(), $request)
+            );
+
+            $this->close($conn, 403);
+            return ;
+        }
+
+        $user = $accessToken->getUser();
+        $token = new AnonymousToken(
+            $request->getQuery()->get('access_token'),
```
Comment (pierre-tranchard) [inline Resources/docs/HandshakeMiddleware.md]: what if the request object is null? and what if the access token is given in header?

File: Server/App/Registry/HandshakeMiddlewareRegistry.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    protected $middlewares;
+
+    public function __construct()
+    {
+        $this->middlewares = [];
+    }
+
+    /**
+     * @param HandshakeMiddlewareAbstract $middleware
+     * @throws \Exception
```
Comment (pierre-tranchard) [inline Server/App/Registry/HandshakeMiddlewareRegistry.php]: where?

**Comment by t3ran13** [comment]
@ProPheT777 
updated, delete composer.lock

@pierre-tranchard 
it is just example 
annotation was fixed

**Comment by t3ran13** [comment]
@ProPheT777 
some news?

**Comment by ProPheT777** [comment]
Yes sorry, I don't have time, I'll try to test this week end, i'm sorry :/

**Comment by t3ran13** [comment]
@ProPheT777 ping

**Pull request closed without merging**